packages/framework/ai-collab/src/explicit-strategy/index.ts

```diff
@@ -0,0 +1,369 @@
+/*!
+ * Copyright (c) Microsoft Corporation and contributors. All rights reserved.
+ * Licensed under the MIT License.
+ */
+
+import {
+	getSimpleSchema,
+	normalizeFieldSchema,
+	Tree,
+	type ImplicitFieldSchema,
+	type SimpleTreeSchema,
+	type TreeNode,
+	type TreeView,
+} from "@fluidframework/tree/internal";
+// eslint-disable-next-line import/no-internal-modules
+import { zodResponseFormat } from "openai/helpers/zod";
+import type {
+	ChatCompletionCreateParams,
+	// eslint-disable-next-line import/no-internal-modules
+} from "openai/resources/index.mjs";
+import { z } from "zod";
+
+import type { OpenAiClientOptions, TokenUsage } from "../aiCollabApi.js";
+
+import { applyAgentEdit } from "./agentEditReducer.js";
+import type { EditWrapper, TreeEdit } from "./agentEditTypes.js";
+import { IdGenerator } from "./idGenerator.js";
+import {
+	getEditingSystemPrompt,
+	getPlanningSystemPrompt,
+	getReviewSystemPrompt,
+	toDecoratedJson,
+	type EditLog,
+} from "./promptGeneration.js";
+import { generateGenericEditTypes } from "./typeGeneration.js";
+import { fail } from "./utils.js";
+
+const DEBUG_LOG: string[] = [];
+
+/**
+ * {@link generateTreeEdits} options.
+ *
+ * @internal
+ */
+export interface GenerateTreeEditsOptions<TSchema extends ImplicitFieldSchema> {
```

Comment:
There seems to be a lot of overlap between this type signature and `AiCollabOptions`. Any way for us to deduplicate?

Reply:
Same with the next couple of interfaces.

Reply:
I wanted to keep them separate to follow in the line of thought that the underlying strategies don't have to conform to the customer API surface. They are very similar at the moment so there is redundancy, but I wanted to keep it this way in the spirit of this concept.